File: ruby/lib/voxgig_station.rb
# frozen_string_literal: true

# voxgig_station - one control surface for outbound integrations.
#
# The Ruby station library (design voxgig/station docs/design/station.md):
# solo mode, hand-written per the modem principle (design 10). sekreto
# (require 'voxgig_sekreto') and struct (require 'voxgig_struct', which
# backs validate_config) are the two dependencies it takes.

require_relative 'voxgig_station/error'
require_relative 'voxgig_station/events'
require_relative 'voxgig_station/descriptor'
require_relative 'voxgig_station/structhome'
require_relative 'voxgig_station/shape'
require_relative 'voxgig_station/feature'
require_relative 'voxgig_station/factory'
require_relative 'voxgig_station/loader'
require_relative 'voxgig_station/profile'
require_relative 'voxgig_station/secrets'
require_relative 'voxgig_station/station'
require_relative 'voxgig_station/adapter'

File: dart/lib/voxgig_station.dart
// voxgig_station - one control surface for outbound integrations.
//
// Dart port of the canonical TypeScript implementation
// (typescript/src/). Solo mode only in v1 (the proxy is a deferred
// amplifier): plugin registry, profiles, env-only secret broker with
// placeholder injection, bounded event ring + tap, descriptor
// normalizer and canonical serializer.

export 'src/adapter.dart'
    show FeatureBinding, StationAdapterFeature, StationTransport,
        adapterFeature, featureBinding;
export 'src/descriptor.dart'
    show Normalized, canonicalSerialize, envtoken, normalizeDescriptor,
        secretnameDefault;
export 'src/error.dart' show CODES, StationError, isKnownCode;
export 'src/events.dart' show EventBuffer, TapFn;
export 'src/profile.dart'
    show MERGE_SENSITIVE, findConfigFile, loadConfig, refapi, resolveProfile,
        selectProfile;
export 'src/secrets.dart' show SecretBroker, envkey, placeholderFor, validname;
export 'src/station.dart' show PluginEntry, RegisterResult, Station;
